### PR TITLE
include rUSD for borrow

### PR DESCRIPTION
**Pull request by eukadish**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/reservoir-protocol exports TVL: 
        
 
 
    ethereum                  20.26 M
    ethereum-borrowed         17.11 M
    borrowed                  17.11 M
    
    total                    20.26 M

File: projects/reservoir-protocol/index.js [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
-    '0x9BB2c38F57883E5285b7c296c66B9eEA4769eF80',
-    '0x99A95a9E38e927486fC878f41Ff8b118Eb632b10',
-    // '0x31Eae643b679A84b37E3d0B4Bd4f5dA90fB04a61', - exluded RUSD because it is project's own token
-  ]
+    ethereum: [
+        // https://docs.reservoir.xyz/products/proof-of-reserves
+        '0x0c7e4342534e6e8783311dCF17828a2aa0951CC7',
+        '0x9BB2c38F57883E5285b7c296c66B9eEA4769eF80',
+        '0x99A95a9E38e927486fC878f41Ff8b118Eb632b10',
+        '0x31Eae643b679A84b37E3d0B4Bd4f5dA90fB04a61'
```
Comment (g1nt0ki) [inline projects/reservoir-protocol/index.js:7]: why do you want to include this here? backing of rUSD is the other vaults right?

File: projects/reservoir-protocol/index.js [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
+            const bals = await api.multiCall({ abi: 'uint256:totalValue', calls: funds })
+            const decimals = await api.multiCall({ abi: 'uint8:decimals', calls: tokens })
+            bals.forEach((v, i) => bals[i] = v * 10 ** (decimals[i] - 18))
+            api.add(tokens, bals)
+
+            const psmBal = await api.call({ abi: 'erc20:balanceOf', target: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48', params: ['0x4809010926aec940b550D34a46A52739f996D75D'] })
+
+            api.add('0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48', psmBal)
+        },
+        borrowed: async (api) => {
```
Comment (g1nt0ki) [inline projects/reservoir-protocol/index.js:25]: atm, we are not showing minted tokens as borrowed, but I have asked another llama to add rUSD to our stablecoin dashboard

File: projects/reservoir-protocol/index.js [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
+    const funds = config[chain]
+    module.exports[chain] = {
+        tvl: async (api) => {
+            const tokens = await api.multiCall({ abi: 'address:underlying', calls: funds })
+            const bals = await api.multiCall({ abi: 'uint256:totalValue', calls: funds })
+            const decimals = await api.multiCall({ abi: 'uint8:decimals', calls: tokens })
+            bals.forEach((v, i) => bals[i] = v * 10 ** (decimals[i] - 18))
+            api.add(tokens, bals)
+
+            const psmBal = await api.call({ abi: 'erc20:balanceOf', target: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48', params: ['0x4809010926aec940b550D34a46A52739f996D75D'] })
```
Comment (g1nt0ki) [inline projects/reservoir-protocol/index.js:21]: added it here: https://github.com/DefiLlama/DefiLlama-Adapters/commit/8b5000b9d808cd5a9fa8536d0d3a26b93530b8d4

File: projects/reservoir-protocol/index.js (hunk 2 above)
Comment (eukadish) [reply projects/reservoir-protocol/index.js:25]: Great, thank you!

File: projects/reservoir-protocol/index.js (hunk 1 above)
Comment (eukadish) [reply projects/reservoir-protocol/index.js:7]: It's all the assets of the protocol, back our liabilities, including rUSD.

Comment (g1nt0ki) [reply projects/reservoir-protocol/index.js:7]: so, counting rUSD as tvl here would be doublecounting?

Comment (eukadish) [reply projects/reservoir-protocol/index.js:7]: Well no, because we add the other side, liability, on as "borrowed".

Comment (g1nt0ki) [reply projects/reservoir-protocol/index.js:7]: sorry, what is backing the rUSD, the assets that we are counting as tvl right?

Comment (eukadish) [reply projects/reservoir-protocol/index.js:7]: The full balance sheet of the protocol, i.e. for our liabilities, all the assets. The RWA loans we issue, our lending are assets within Reservoir.

File: projects/reservoir-protocol/index.js (hunk 3 above)
Comment (eukadish) [reply projects/reservoir-protocol/index.js:21]: Will this PR get merged? Or are the changes simply in the commit you shared?

Comment (eukadish) [reply projects/reservoir-protocol/index.js:21]: @g1nt0ki are there any outstanding issues or can this be merged?

File: projects/reservoir-protocol/index.js (hunk 1 above)
Comment (eukadish) [reply projects/reservoir-protocol/index.js:7]: It's that our debt is offset by assets, so we want to have a way to show our liabilities in that the value is offset by the total assets held in by the protocol.

Comment (eukadish) [reply projects/reservoir-protocol/index.js:7]: We want to add all the additional asset we have, so will offset by the liabilities we have.

Comment (g1nt0ki) [reply projects/reservoir-protocol/index.js:7]: @eukadish if the assets are not locked in your contracts, liabilities or not, we can track them only under borrowed.

Comment (g1nt0ki) [reply projects/reservoir-protocol/index.js:7]: the reason we are stubborn about this is, we have had issues where protocols were inflating tvl by cyclic lending

Comment (eukadish) [reply projects/reservoir-protocol/index.js:7]: Understood, understood. Okay, then let me sync with my team and see if we can break it down better or maybe change how we represent our numbers and get back to you.

Comment (g1nt0ki) [reply projects/reservoir-protocol/index.js:7]: hi @eukadish I am closing this for now, we can reopen the PR later

**Pull request closed without merging**